lib/react.dart

```diff
@@ -775,6 +829,44 @@ abstract class Component2 implements Component {
   List<StateUpdaterCallback> get transactionalSetStateCallbacks => throw new UnimplementedError();
 }
 
+/// Creates a mixin used to ensure consistent typing of the `snapshot`
+/// parameter passed between Component2.getSnapshotBeforeUpdate and
+/// Component2.componentWillUpdate.
+///
+/// __EXAMPLE__:
+///
+/// class _ParentComponent extends react.Component2 implements react
+///    .TypedSnapshot<int> {
```

Comment:
Could we put this on the previous line, or insert a newline `implements` instead?